src/ILCompiler.Compiler/src/Compiler/VirtualMethodCallHelper.cs

```diff
@@ -83,7 +83,11 @@ public static int GetGenericDictionarySlot(NodeFactory factory, TypeDesc type)
         /// </summary>
         public static bool HasGenericDictionarySlot(this TypeDesc type)
         {
-            return !type.IsInterface && type.HasInstantiation &&
+            // Dictionary slots on generic interfaces are necessary to support static methods on interfaces
+            if (type.IsInterface)
```

Comment:
The dictionary slot is unconditionally created on generic interfaces, regardless of whether they share code or not.
See code/comment in TypeLoaderEnvironment.Metadata.cs:
`                        if (RuntimeAugments.IsGenericType(searchForSharedGenericTypesInParentHierarchy))
                        {
                            if (RuntimeAugments.IsInterface(searchForSharedGenericTypesInParentHierarchy))
                            {
                                // Generic interfaces always have a dictionary slot in the vtable (see binder code in MdilModule::SaveMethodTable)
                                // Interfaces do not have base types, so we can just break out of the loop here ...
                                slot++;
                                break;
                            }`